### DISCUSSION TITLE
Ideas: Topics to cover

### DISCUSSION THREAD

**Discussion by dev-experience** [Extenza-Academy/WebDev-100_2021-Q1#14] [Ideas]
- [ ] Webpack

**Comment by raybags-dev** [comment]
**Project preposition**

Hi **@mserykh**  and **@arkhipovilia01** I hope you guys are doing alright given the current covid crisis. I'm proposing a project I'd call  the **"Album".**  This project would ground our knowledge  in the most common concepts in front end development like  

- DOM events https://developer.mozilla.org/en-US/docs/Web/Events

- Event listeners https://developer.mozilla.org/en-US/docs/Web/API/EventListener

- Event throtaling and bubling https://css-tricks.com/debouncing-throttling-explained-examples/

- Click events, 

- Responsive images https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images

- Lazy loading https://developer.mozilla.org/en-us/docs/Glossary/Lazy_load

- Javascript array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

- Principles like:- Single responsibility principle  in functional programming (not in classical programing as i beleive some of us have not tackled classes in js  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes. If you have, we can work with classes as welll). 

We will also work with higher order functions https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions 

We can also work with CSS principles like Flex-box t.ly/ZChJ or grid, through i always prefer flex. Its just awesome, 

We will also impliment Media queries for responsive screen sizes, orientation and resizes depending on "mobile-first" or "desk-top first" . 

I know you guys are working hard and busy with your own studies , if  this is not a good time, its alright, kjust let me know. 

Please let me know if you are interested so we can discuss the project layout.
